### PR TITLE
FIX from N queries to 1 query at csubs collection upon context element processing (N: number of attributes)

### PR DESCRIPTION
**Pull request by fgalan**
It completes (or not? :) issue #1475 

Unit test: all working except the following ones (I will have a look in parallel to @kzangeli ):

```
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] mongoUpdateContext_withOnchangeSubscriptions.Cond1_deleteMixMatchNoMatch
[  FAILED  ] mongoUpdateContext_withOnchangeSubscriptions.CondN_deleteMixMatchNoMatch
```

Test harness: run using `-noCache` (as the cache way is not working due to DIRTY_HACK) all them working, except some ones expected to fail:

```
----------- Failing tests ------------------
  o  0000_statistics_operation/statistics.test
  o  0000_statistics_operation/statistics_with_semaphores.test
  o  0563_ngsi10_get_list_of_all_entities/all_entities_no_filter.test
  o  0581_include_type_in_convops/v1_registry_contextEntities_enType_ENTYPE_enId_ENID_attribute_attrName.test
  o  1048_subscription_cache/cache_population_at_init.test
  o  1048_subscription_cache/three_subscriptions.test
  o  1308_subscription_cache/subCacheMixedSubscriptionsWithUpdatesAndDeletes.test
  o  1308_subscription_cache/subCacheThrottlingAndExpirationAfterRefresh.test
  o  1308_subscription_cache/sub_expired_and_updated_then_cache_refresh.test
  o  1367_time_measures/time_measure_stats.test
```

(I havent expected v1_registry_contextEntities_enType_ENTYPE_enId_ENID_attribute_attrName.test to fail, maybe it has been a glitch)

### CODE REVIEW

File: test/functionalTest/harnessFunctions.sh [hunk 1]
```diff
@@ -341,32 +341,32 @@ function localBrokerStart()
   if [ "$role" == "CB" ]
   then
     port=$CB_PORT
-    CB_START_CMD="contextBroker -harakiri -port $CB_PORT  -pidpath $CB_PID_FILE  -dbhost $dbHost:$dbPort -db $CB_DB_NAME -dbPoolSize $POOL_SIZE -t $traceLevels $IPvOption $extraParams"
+    CB_START_CMD="contextBroker -noCache -harakiri -port $CB_PORT  -pidpath $CB_PID_FILE  -dbhost $dbHost:$dbPort -db $CB_DB_NAME -dbPoolSize $POOL_SIZE -t $traceLevels $IPvOption $extraParams"
```
Comment (fgalan) [inline test/functionalTest/harnessFunctions.sh]: Changes in this file should be reverted before PR is merged (`-noCahe` was used to test while DIRTY_HACK was in place).

File: src/lib/ngsi9/DiscoverContextAvailabilityResponse.cpp [hunk 2]
```diff
@@ -101,15 +101,6 @@ std::string DiscoverContextAvailabilityResponse::render(RequestType requestType,
       out += errorCode.render(format, indent + "  ");
   }
 
-#if 0
```
Comment (fgalan) [inline src/lib/ngsi9/DiscoverContextAvailabilityResponse.cpp]: Not related with this PR, but spoted and agreed with @kzangeli to remove it. NTC.

**Comment by kzangeli** [comment]
LGTM

**Comment by fgalan** [comment]
LGTM

**Pull request merged**